### DISCUSSION TITLE
Add Raven-Clr-Type column in Documents view in Studio

### DISCUSSION THREAD

**Discussion by mic2zar** [ravendb/ravendb#15486] [Q&A] [answered]
I am in the process of redefining parts of our system to use event sourcing approach. To achieve this I decided (following your docs and webinar) for many small event type classes, which I all store afterwards in one big collection (I mean Oren, in his webinar about event sourcing stored two of his event types in two separate collection, but in my opinion for anything bigger than trivial, we would need to finish with tenths of additional collections, making it very challenging to query; also cluttering the view in the database). 

Now I would like to see my events by types. The default Document and Collections view in Studio does not show Raven-Clr-Type. I tried to follow your documentation about this topic [1], but also there I could not find any solution. I tried adding anything like:
this.[@metadata][@Raven-Clr-Type]
[@metadata][@Raven-Clr-Type]
@metadata.Raven-Clr-Type
and many others, but each of them is reporting mapping error. Would it be even possible? Maybe my design of event sourcing should be approached differently? 

[1] https://ravendb.net/docs/article-page/5.4/csharp/studio/database/documents/documents-and-collections#the-documents-view

**Comment by garayx** [comment, marked as answer]
Hi

You can add the `RavenClrType` custom column to the studio by using `this.__metadata.ravenClrType`.

![image](https://user-images.githubusercontent.com/16625387/209104319-74668b48-8f52-4b17-8f6c-e4082889c40c.png)

**Reply by mic2zar** [reply]
Excellent. This is exactly what I was looking for. 

Is there any way to extract only relevant info (type) from this string already in Studio or do I need to write a custom index for that? Something along this line:
this.__metadata.ravenClrType.Replace(', redi-backend', '')

**Reply by ayende** [reply]
You can put any code there.

**Reply by mic2zar** [reply]
aahh. I see, in jscript. Here is my solution in case somebody would need this in the future:
this.__metadata.ravenClrType.replace(", xxx", "").replace("yyy", "")

thank you

**Reply by Danielle9897** [reply]
added issue:
https://issues.hibernatingrhinos.com/issue/RDoc-2340/Add-explanation-about-the-custom-column-that-can-be-added-to-the-display
